libs/langgraph/langgraph/pregel/__init__.py

```diff
@@ -629,6 +634,7 @@ async def aupdate_state(
         # get last checkpoint
         saved = await self.checkpointer.aget_tuple(config)
         checkpoint = copy_checkpoint(saved.checkpoint) if saved else empty_checkpoint()
+        checkpoint_previous_versions = checkpoint["channel_versions"].copy()
```

Comment:
Why is this being copied?

Reply:
cause it's mutated below, but i updated to avoid copying altogether